elfo-core/src/messages.rs

```diff
@@ -52,6 +52,28 @@ impl<R: Display> From<R> for ConfigRejected {
     }
 }
 
+#[message(ret = Result<(), EntrypointFailed>)]
+#[derive(Constructor)]
+#[non_exhaustive]
+pub struct StartEntrypoint {
+    pub is_check_only: bool,
+}
+
+#[message]
```

Comment:
It's better to use `message(part)` here to avoid adding to the message index. It's possible because `EntrypointFailed` isn't sent directly.